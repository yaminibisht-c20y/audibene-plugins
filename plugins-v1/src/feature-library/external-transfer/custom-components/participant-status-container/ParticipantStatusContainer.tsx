import * as React from 'react';
import styled from 'styled-components';
import {
  withTheme,
} from '@twilio/flex-ui';
import ParticipantName from '../participant-name';
import ParticipantStatus from '../participant-status';

const StatusContainer = styled.div`
  display: flex;
  flex-wrap: nowrap;
  flex-grow: 1;
  flex-shrink: 1;
  flex-direction: column;
  overflow: hidden;
`;

class ParticipantStatusContainer extends React.PureComponent <any,any> {
  render() {
    return (
      <div>
        <ParticipantName key="custom-name" {...this.props} />
        <ParticipantStatus key="custom-status" {...this.props} />
      </div>
    );
  }
}

export default withTheme(ParticipantStatusContainer);