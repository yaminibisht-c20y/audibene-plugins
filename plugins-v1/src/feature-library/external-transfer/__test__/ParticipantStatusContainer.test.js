import React from 'react';
import { shallow } from 'enzyme';
import ParticipantStatusContainer from "../custom-components/participant-status-container";

jest.mock("@twilio/flex-ui",()=>({
    withTheme: (M) =>{ return M},
}))

describe("test ParticipantStatusContainer component", () => {
    let props = {
        task:{
            conference:{
                participants:[{status: "joined"}]
            },
            attributes:{
                outbound_to:""
            }
        },
        theme:{
            CallCanvas:{
                Button: false
            }
        },
        isOpen: true,
        participant:{
            callSid: "",
            participantType: "customer",
            onHold: true
        }
    }
    it("component should render successfully", () => {
        const wrapper = shallow(<ParticipantStatusContainer {...props}/>);
        expect(wrapper).toMatchSnapshot();
    })
})