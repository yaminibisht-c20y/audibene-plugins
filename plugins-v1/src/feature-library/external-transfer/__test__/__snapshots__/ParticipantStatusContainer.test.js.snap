// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ParticipantStatusContainer component component should render successfully 1`] = `ShallowWrapper {}`;
